next/src/types.ts

```diff
@@ -36,3 +48,11 @@ export type JsfSchema = JSONSchema & {
  * @see `JsfSchema` for the full schema type which allows booleans and is used for sub schemas.
  */
 export type NonBooleanJsfSchema = Exclude<JsfSchema, boolean>
+
+/**
+ * JSON Schema Form type specifically for object schemas.
+ * This type ensures the schema has type 'object'.
+ */
+export type JsfObjectSchema = NonBooleanJsfSchema & {
+  type: 'object'
+}
```

Comment:
I've introduced this type so we can enforce at the type level schemas being passed to `createHeadlessForm` and certain other functions are object schemas.